### PR TITLE
optimize transaction route

### PR DESCRIPTION
**Pull request by Qiao-Jin**
Remake of #2116, which can help stabilize TPS:

image

### CODE REVIEW

File: src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs [hunk 1]
```diff
@@ -294,15 +295,23 @@ private void OnHeadersMessageReceived(HeadersPayload payload)
...
         {
+            knownHashes.Add(inventory.Hash);
             pendingKnownHashes.Remove(inventory.Hash);
-            if (inventory is Block block)
+            switch (inventory)
             {
-                UpdateLastBlockIndex(block.Index);
-                if (block.Index > NativeContract.Ledger.CurrentIndex(system.StoreView) + InvPayload.MaxHashesCount) return;
+                case Transaction transaction:
+                    system.TxRouter.Forward(new TransactionRouter.Preverify(transaction, true));
```
Comment (erikzhang) [inline src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs:303]: Why `Forward`? I think it should be `Tell`.

Comment (Qiao-Jin) [reply src/neo/Network/P2P/RemoteNode.ProtocolHandler.cs]: Have fixed

File: src/neo/Ledger/Blockchain.cs [hunk 2]
```diff
@@ -298,14 +293,6 @@ protected override void OnReceive(object message)
             }
         }
 
-        private void OnTransaction(Transaction tx)
-        {
-            if (ContainsTransaction(tx.Hash))
```
Comment (erikzhang) [inline src/neo/Ledger/Blockchain.cs:303]: This is removed. But I think we should check it first before sending the transaction to `txrouter`.

Comment (Qiao-Jin) [reply src/neo/Ledger/Blockchain.cs:303]: Maybe we can check this in RemoteNode rather than in Blockchain to avoid message hoop

Comment (erikzhang) [reply src/neo/Ledger/Blockchain.cs:303]: We can check it in `RemoteNode` or `TransactionRouter`.

Comment (Qiao-Jin) [reply src/neo/Ledger/Blockchain.cs:303]: Have added

**Comment by erikzhang** [comment]
I think `Blockchain` should keep the logic of processing transactions. Because the wallets may directly send the signed transaction to the `Blockchain`.

**Comment by Qiao-Jin** [comment]
> I think `Blockchain` should keep the logic of processing transactions. Because the wallets may directly send the signed transaction to the `Blockchain`.

Have fixed

**Review by erikzhang — APPROVED** [review]

**Review by shargon — APPROVED** [review]

**Pull request merged**